spec/components/admin/schools/cohorts/other_info_spec.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+RSpec.describe Admin::Schools::Cohorts::OtherInfo, type: :view_component do
+  let(:cohort) { instance_double Cohort, start_year: rand(2020..2030) }
+  let(:school_cohort) { instance_double(SchoolCohort, induction_programme_choice: Faker::Lorem.words.join("_")) if rand < 0.5 }
+
+  component { described_class.new cohort: cohort, school_cohort: school_cohort }
+
+  before do
+    url_options[:school_id] = 1
+    allow(controller).to receive(:url_options).and_return(controller.url_options.merge(school_id: 1))
+  end
+
+  context "with admin_change_programme feature enabled", with_feature_flags: { admin_change_programme: "active" } do
+    it { is_expected.to have_link "Change", href: admin_school_change_programme_path(id: cohort.start_year) }
+  end
+
+  context "with admin_change_programme feature disabled", with_feature_flags: { admin_change_programme: "inactive" } do
+    it { is_expected.not_to have_link "Change" }
+  end
+
+  context "without school cohort" do
+    let(:school_cohort) { nil }
+
+    it { is_expected.to have_content "No programme" }
+  end
+
+  context "with design your own school cohort" do
+    let(:school_cohort) { instance_double SchoolCohort, induction_programme_choice: "design_your_own" }
```

Comment:
If we had used a real class here this would have caught the `design_your_own` vs `design_our_own` issue